workout.py

```diff
@@ -35,29 +114,39 @@ def handle_workout_search():
     )
     json_data = response.json()
     workouts_obj = json_data
-    return render_template('display_options.html', results = workouts_obj)
+    return render_template('display_options.html', results = workouts_obj,
+    target_area=form_data['targetArea'])
 
-@app.route('/handle_workout_selection', methods=['GET', 'POST'])
-def handle_workout_selection():
-    '''Handle a users selection of workouts'''
+@app.route('/handle_workout_submission', methods=['GET', 'POST'])
+@login_required
+def handle_workout_submission():
+    '''Handle a users report of their workout'''
     form_data = request.form
-    api_url = 'https://exercises-by-api-ninjas.p.rapidapi.com/v1/exercises'
-    api_key = os.getenv('X_API_KEY')
-    headers = {
-	"X-RapidAPI-Key": api_key,
-	"X-RapidAPI-Host": "exercises-by-api-ninjas.p.rapidapi.com"
-    }
-    querystring = {'name': form_data['chosenWorkout']}
-    response = requests.get(
-        api_url,
-        headers=headers,
-        params=querystring
-    )
-    json_data = response.json()
-    workout_obj = json_data
-    chosen_workout = ""
-    for workout in workout_obj:
-        if str(form_data['chosenWorkout']) == str(workout['name']):
-            chosen_workout = workout['name']
-            return render_template('show_chosen_workout.html', workout = chosen_workout)
+    num_sets = form_data['sets']
+    target_area = form_data['target_area']
+    todays_workout = workout['today']
+    workout['today'] = f'{target_area} sets: {num_sets}. {todays_workout}'
+    flash(f'Added {target_area} to this workout.')
+    return render_template('continue_end.html')
+
+@app.route('/continue_response', methods=['POST'])
+@login_required
+def continue_response():
+    '''The user opted to continue the workout session'''
+    return render_template('multitarget_workout.html')
+
+@app.route('/end_workout', methods=['POST'])
+@login_required
+def end_workout():
```

Comment:
I followed the logic all the way through adding workouts and didn't find any errors. Target muscle group and # of sets are added in a string to the db along with username and date to be queried by for muscle group suggestion.